File: AkhterAirPlayReceiver/app/src/main/java/com/github/serezhka/jap2server/AirplayDataConsumer.java
package com.github.serezhka.jap2server;

import com.github.serezhka.jap2lib.rtsp.AudioStreamInfo;
import com.github.serezhka.jap2lib.rtsp.VideoStreamInfo;

public interface AirplayDataConsumer {

    void onVideo(byte[] video, long timestampUs, long sequenceNumber, boolean codecConfig);

    void onVideoFormat(VideoStreamInfo videoStreamInfo);

    void onAudio(byte[] audio, long timestamp, int sequenceNumber);

    void onAudioFormat(AudioStreamInfo audioInfo);
}

File: airplaytestlab/app/src/main/java/com/akhter/airplaytestlab/apserver/AirplayDataConsumer.kt
package com.akhter.airplaytestlab.apserver

import com.akhter.airplaytestlab.aplib.rtsp.AudioStreamInfo
import com.akhter.airplaytestlab.aplib.rtsp.VideoStreamInfo

interface AirplayDataConsumer {

    fun onVideo(video: ByteArray?)

    fun onVideoFormat(videoStreamInfo: VideoStreamInfo?)

    fun onAudio(audio: ByteArray?)

    fun onAudioFormat(audioInfo: AudioStreamInfo?)
}